include/work_queue.h

```diff
@@ -22,7 +22,7 @@ class WorkQueue {
     friend class WorkQueue;
    public:
     node_id_t get_node_idx() { return node_idx; }
-    std::vector<node_id_t> get_data_vec() { return data_vec; }
+    std::vector<node_id_t>& get_data_vec() { return data_vec; }
```

Comment:
Good change, lets make it a const reference though so that the caller cannot do bad things like modify the size of the vector for example.